src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateAttributeDefinitionActionUtils.java

```diff
@@ -0,0 +1,278 @@
+     *
+     * @param oldAttributeDefinitions                       the old list of attribute definitions.
+     * @param newAttributeDefinitionsDrafts                 the new list of attribute definitions drafts.
+     * @param syncOptions                                   the sync options with which a custom callback function is
+     *                                                      called if warnings or errors.
+     * @return a list of attribute definitions update actions if the list of attribute definitions is not identical.
+     *         Otherwise, if the attribute definitions are identical, an empty list is returned.
+     * @throws BuildUpdateActionException in case there are attribute definitions drafts with duplicate names.
+     */
+    @Nonnull
+    public static List<UpdateAction<ProductType>> buildAttributeDefinitionsUpdateActions(
+        @Nonnull final List<AttributeDefinition> oldAttributeDefinitions,
+        @Nullable final List<AttributeDefinitionDraft> newAttributeDefinitionsDrafts,
+        @Nonnull final ProductTypeSyncOptions syncOptions)
+        throws BuildUpdateActionException {
+
+        if (newAttributeDefinitionsDrafts != null) {
+            return buildUpdateActions(
+                oldAttributeDefinitions,
+                newAttributeDefinitionsDrafts,
+                syncOptions
+            );
+        } else {
+            return oldAttributeDefinitions
+                .stream()
+                .map(AttributeDefinition::getName)
+                .map(RemoveAttributeDefinition::of)
+                .collect(Collectors.toList());
+        }
+    }
+
+    /**
+     * Compares a list of {@link AttributeDefinition}s with a list of {@link AttributeDefinitionDraft}s.
+     * The method serves as an implementation for attribute definitions syncing. The method takes in functions
+     * for building the required update actions (AddAttribute, RemoveAttribute, ChangeAttributeOrder and 1-1
+     * update actions on attribute definitions (e.g. changeAttributeName, changeAttributeLabel, etc..) for the required
+     * resource.
+     *
+     * @param oldAttributeDefinitions                       the old list of attribute definitions.
+     * @param newAttributeDefinitionsDrafts                 the new list of attribute definitions drafts.
+     * @param syncOptions                                   the sync options with which a custom callback function is
+     *                                                      called if warnings or errors.
+     *
+     * @return a list of attribute definitions update actions if the list of attribute definitions is not identical.
+     *         Otherwise, if the attribute definitions are identical, an empty list is returned.
+     * @throws BuildUpdateActionException in case there are attribute definitions drafts with duplicate names.
+     */
+    @Nonnull
+    private static List<UpdateAction<ProductType>>
+        buildUpdateActions(
```

Comment:
I think the indentation of params like the previous method is much easier to read (it's also the same across the whole lib)